bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/network/SiemensHvacConnectorImpl.java

```diff
@@ -0,0 +1,489 @@
+                                            sessionId = resultObj.get("SessionId").getAsString();
+                                            logger.debug("Have new SessionId : {} ", sessionId);
+                                        }
+
+                                    }
+
+                                }
+                            }
+
+                            logger.debug("siemensHvac:doAuth:decodeResponse:()");
+
+                            if (sessionId == null) {
+                                logger.debug("Session request auth was unsucessfull in _doAuth()");
+                            }
+
+                        }
+
+                    }
+                }
+            }
+
+            logger.debug("siemensHvac:doAuth:connect()");
+
+        } catch (Exception ex) {
+            logger.debug("siemensHvac:doAuth:error() {}", ex.getLocalizedMessage());
+        } finally {
+        }
+    }
+
+    public @Nullable String DoBasicRequest(String uri) throws Exception {
+        return DoBasicRequest(uri, null);
+    }
+
+    public @Nullable String DoBasicRequestAsync(String uri, @Nullable SiemensHvacCallback callback) throws Exception {
+        return DoBasicRequest(uri, callback);
+    }
+
+    public @Nullable String DoBasicRequest(String uri, @Nullable SiemensHvacCallback callback) throws Exception {
+        if (sessionIdHttp == null) {
+            doAuth(true);
+        }
+
+        if (sessionId == null) {
+            doAuth(false);
+        }
+
+        try {
+            String baseUri = baseUrl;
+
+            String mUri = uri;
+            if (!mUri.endsWith("?")) {
+                mUri = mUri + "&";
+            }
+            if (mUri.indexOf("main.app") >= 0) {
+                mUri = mUri + "SessionId=" + sessionIdHttp;
+            } else {
+                mUri = mUri + "SessionId=" + sessionId;
+            }
+
+            logger.debug("Execute request: {}", uri);
+            CookieStore c = httpClient.getCookieStore();
+            java.net.HttpCookie cookie = new HttpCookie("SessionId", sessionIdHttp);
+            cookie.setPath("/");
+            cookie.setVersion(0);
+
+            c.add(new URI(baseUri), cookie);
+
+            logger.debug("Execute request: {}", uri);
+            final Request request = httpClient.newRequest(baseUri + mUri);
+            request.method(HttpMethod.GET);
+
+            ContentResponse response = executeRequest(request, callback);
+            if (callback == null && response != null) {
+                int statusCode = response.getStatus();
+
+                if (statusCode == HttpStatus.OK_200) {
+                    String result = response.getContentAsString();
+
+                    return result;
+                }
+            }
+        } catch (Exception ex) {
+            logger.error("siemensHvac:DoRequest:Exception by executing Request: {} ; {} ", uri,
+                    ex.getLocalizedMessage());
```

Comment:
Unless this will stop the binding from working:
```suggestion
            logger.warn("siemensHvac:DoRequest:Exception by executing Request: {}: {} ", uri,
                    ex.getLocalizedMessage());
```

Since warnings will be shown to the user, you might want to format this differently/less technical. Or further reduce to debug.

Perhaps it would be better to throw an exception, so that the caller could also set thing status to OFFLINE with COMMUNICATION_ERROR.

Reply:
I move it to warn.
There could be transiant exception when the gateway is overload, but the binding will recover itself at next request.